Automattic/wp-calypso
# Framework: Add current user handlers to Redux state

This pull request seeks to add current user handlers to the Redux state tree. As of this pull request, the current user is not used anywhere, but will be used to replace the need for a [`currentUserID` prop](https://github.com/Automattic/wp-calypso/blob/fb1df04cd9d65ff7be4a238b415bd93270feebb1/client/post-editor/editor-sharing/container.jsx#L85) to the `<SharingConnectionsContainer />` component, among many other expected use-cases.

__Implementation notes:__

The intent and implementation closely resembles that of the currently selected site, so you can reference the similarities in your review.

I chose to suffix action imperatives in the naming of the action types, e.g. `USER_RECEIVE` instead of `RECEIVE_USER` per outcome of [some previous discussion](https://github.com/Automattic/wp-calypso/pull/1660/files#r47770586) on maintainability of alphabetized action types.

__Testing instructions:__

This will be easiest to test with the [Redux DevTools Chrome plugin](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd), for which support was recently added in #2012.

1. Navigate to any Calypso page
2. Click the Redux DevTools icon in the address bar
3. Scroll to the bottom of the displayed overlay
4. Note that the final state includes an indexed user object for your user account under `users` and `currentUser` ID under `ui`

Also confirm that Mocha tests pass by running `make test` from the project root directory or from both `client/state/ui` and `client/state/users`.
